apis/container/v1beta1/zz_cluster_types.go

```diff
@@ -4603,8 +4603,18 @@ type PrivateClusterConfigObservation struct {
 	PrivateEndpoint *string `json:"privateEndpoint,omitempty" tf:"private_endpoint,omitempty"`
 
 	// Subnetwork in cluster's network where master's endpoint will be provisioned.
+	// +crossplane:generate:reference:type=github.com/upbound/provider-gcp/apis/compute/v1beta2.Subnetwork
+	// +crossplane:generate:reference:extractor=github.com/upbound/provider-gcp/config/common.SelfLinkExtractor()
 	PrivateEndpointSubnetwork *string `json:"privateEndpointSubnetwork,omitempty" tf:"private_endpoint_subnetwork,omitempty"`
 
+	// Reference to a Subnetwork in compute to populate privateEndpointSubnetwork.
+	// +kubebuilder:validation:Optional
+	PrivateEndpointSubnetworkRef *v1.Reference `json:"privateEndpointSubnetworkRef,omitempty" tf:"-"`
+
+	// Selector for a Subnetwork in compute to populate privateEndpointSubnetwork.
+	// +kubebuilder:validation:Optional
+	PrivateEndpointSubnetworkSelector *v1.Selector `json:"privateEndpointSubnetworkSelector,omitempty" tf:"-"`
+
```

Comment:
I see that these changes are made in PrivateClusterConfigInitParameters in `v1beta2` file, can you please move it from observation to init parameters?